### PR TITLE
Add API support for PI endpoints

### PR DESCRIPTION
**Pull request by zigdon**
Kinda barebones, and would consider adjusting some of the names (maybe facilities instead of pins?) but figured I'd start with CCPs terminology.

### CODE REVIEW

File: evelink/char.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 from evelink.parsing.contract_items import parse_contract_items
 from evelink.parsing.contracts import parse_contracts
 from evelink.parsing.industry_jobs import parse_industry_jobs
+from evelink.parsing.planetary_interactions import parse_planetary_colonies, parse_planetary_links, parse_planetary_pins, parse_planetary_routes
```
Comment (ayust) [inline evelink/char.py]: So there are examples of both ways in EVELink (sadly), but I'd prefer to go without comma imports. Can you split this into 4 import lines please?

File: evelink/parsing/planetary_interactions.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
+from evelink import api
+
+def parse_planetary_colonies(api_results):
+    result = {}
+    for rowset in api_results.findall('rowset'):
+        if rowset is None:
```
Comment (ayust) [inline evelink/parsing/planetary_interactions.py]: What's this about?

File: evelink/parsing/planetary_interactions.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+                'number_of_pins': int(a['numberOfPins']),
+            }
+
+    return result
+
+
+def parse_planetary_links(api_results):
+    result = {}
+    for rowset in api_results.findall('rowset'):
+        if rowset is None:
```
Comment (ayust) [inline evelink/parsing/planetary_interactions.py]: ditto

File: evelink/parsing/planetary_interactions.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+                'destination_id': int(a['destinationPinID']),
+                'link_level': int(a['linkLevel']),
+            }
+
+    return result
+
+def parse_planetary_pins(api_results):
+    result = {}
+    for rowset in api_results.findall('rowset'):
+        if rowset is None:
```
Comment (ayust) [inline evelink/parsing/planetary_interactions.py]: ditto

File: evelink/parsing/planetary_interactions.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+                'loc': {'long': float(a['longitude']),
+                        'lat': float(a['latitude'])},
+            }
+
+    return result
+
+def parse_planetary_routes(api_results):
+    result = {}
+    for rowset in api_results.findall('rowset'):
+        if rowset is None:
```
Comment (ayust) [inline evelink/parsing/planetary_interactions.py]: ditto

File: evelink/parsing/planetary_interactions.py [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+        for row in rowset.findall('row'):
+            # shortcut to make the following block less painful
+            a = row.attrib
+            planetID = int(a['planetID'])
+            result[planetID] = {
+                'id': planetID,
+                'system': {
+                    'id': int(a['solarSystemID']),
+                    'name': a['solarSystemName']
```
Comment (ayust) [inline evelink/parsing/planetary_interactions.py]: trailing commas please

File: evelink/parsing/planetary_interactions.py [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+                'cycle_time': int(a['cycleTime']),
+                'quantity_per_cycle': int(a['quantityPerCycle']),
+                'install_ts': api.parse_ts(a['installTime']),
+                'expiry_ts': api.parse_ts(a['expiryTime']),
+                'content': {
+                    'type': int(a['contentTypeID']),
+                    'name': a['contentTypeName'],
+                    'quantity': int(a['contentQuantity']),
+                },
+                'loc': {'long': float(a['longitude']),
```
Comment (ayust) [inline evelink/parsing/planetary_interactions.py]: Fix indent/wrapping

File: evelink/parsing/planetary_interactions.py [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+            result[routeID] = {
+                'id': routeID,
+                'source_id': int(a['sourcePinID']),
+                'destination_id': int(a['destinationPinID']),
+                'content': {
+                    'type': int(a['contentTypeID']),
+                    'name': a['contentTypeName'],
+                },
+                'quantity': int(a['quantity']),
+                'path': (int(a['waypoint1']), int(a['waypoint2']),
```
Comment (ayust) [inline evelink/parsing/planetary_interactions.py]: Perhaps...

    'path': tuple(int(a['waypoint%d' % n]) for n in range(1,6)),

File: evelink/parsing/planetary_interactions.py (hunk 2 above)
Comment (zigdon) [inline evelink/parsing/planetary_interactions.py]: I think I was worried at what happens if you have no colonies? But doesn't seem to break without it.

File: evelink/parsing/planetary_interactions.py [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+                'install_ts': api.parse_ts(a['installTime']),
+                'expiry_ts': api.parse_ts(a['expiryTime']),
+                'content': {
+                    'type': int(a['contentTypeID']),
+                    'name': a['contentTypeName'],
+                    'quantity': int(a['contentQuantity']),
+                },
+                'loc': {
+                    'long': float(a['longitude']),
+                    'lat': float(a['latitude'])
```
Comment (ayust) [inline evelink/parsing/planetary_interactions.py]: trailing comma

**Pull request merged**